### ISSUE TITLE
Interaction with a closed file should raise a ValueError (not OSError)

### ISSUE THREAD

**Issue by ntoll** [bbcmicrobit/micropython#278]
As per core Python docs: https://docs.python.org/3.5/library/io.html#io.IOBase.close

Example of current behaviour below:

```
>>> x = open('foo.txt')
>>> x.close()
>>> x.read(1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OSError: 9
```

**Comment by dpgeorge** [comment]
See #280.

**Comment by markshannon** [comment]
Fixed by #280
